tests/apollo/test_skvbc_reconfiguration.py

```diff
@@ -489,6 +489,66 @@ async def test_pruning_command(self, bft_network):
                 pruned_block = int(data.additional_data.decode('utf-8'))
                 assert pruned_block <= 90
 
+    @with_trio
+    @with_bft_network(start_replica_cmd, selected_configs=lambda n, f, c: n == 7)
+    async def test_pruning_command_with_failures(self, bft_network):
+        with log.start_action(action_type="test_pruning_command_with_faliures"):
+            bft_network.start_all_replicas()
+            skvbc = kvbc.SimpleKVBCProtocol(bft_network)
+            client = bft_network.random_client()
+
+            # Create 100 blocks in total, including the genesis block we have 101 blocks
+            k, v = await skvbc.write_known_kv()
+            for i in range(99):
+                v = skvbc.random_value()
+                await client.write(skvbc.write_req([], [(k, v)], 0))
+
+            # Get the minimal latest pruneable block among all replicas
+            op = operator.Operator(bft_network.config, client,  bft_network.builddir)
+            await op.latest_pruneable_block()
+
+            latest_pruneable_blocks = []
+            rsi_rep = client.get_rsi_replies()
+            for r in rsi_rep.values():
+                lpab = cmf_msgs.ReconfigurationResponse.deserialize(r)[0]
+                latest_pruneable_blocks += [lpab.response]
+
+            # Now, crash on of the non-primary replicas
+            crashed_replica = 3
+            bft_network.stop_replica(crashed_replica)
+            await op.prune(latest_pruneable_blocks)
+            rsi_rep = client.get_rsi_replies()
+            # we expect to have at least 2f + 1 replies
+            for rep in rsi_rep:
+                r = rsi_rep[rep]
+                data = cmf_msgs.ReconfigurationResponse.deserialize(r)[0]
+                pruned_block = int(data.additional_data.decode('utf-8'))
+                assert pruned_block <= 90
```

Comment:
Why do we assert for less than or equal to 90?

Reply:
because we create at most 90 blocks at the beginning of the test

Reply:
Don't we create 100 blocks in the beginning? Also, how many are expected to be pruned?

Reply:
Oh, sorry. you right,
We create 100 and the default blocks to keep is 10 (in https://github.com/vmware/concord-bft/blob/master/tests/simpleKVBC/TesterReplica/setup.cpp#L60)

Reply:
Thanks!